### PR TITLE
[util/ci] Automatically run MUBI generation

### PR DESCRIPTION
**Pull request by rswarbrick**
It's possible I'm just being stupid, but I couldn't work out how to run the MUBI generation script. I also don't think they're wired into CI. This PR has two patches. The first adds some paths to the generation script so that it would run :-) The second runs it in CI.

@msfschaffner: Have I missed something obvious? If so, I'm very happy to drop this PR and do it the Right Way :-D

### CODE REVIEW

File: hw/ip/prim/util/generate_prim_mubi.py [hunk 1]
```diff
@@ -23,14 +25,22 @@
 
 def main():
 
+    this_dir = os.path.dirname(__file__)
+    repo_top = os.path.normpath(os.path.join(this_dir, '../../../..'))
```
Comment (mcy) [inline hw/ip/prim/util/generate_prim_mubi.py:29]: I recommend:
- using `pathlib`.
- making `REPO_TOP` a global, as a lot of other scripts do.

**Review by mcy — APPROVED** [review]

**Comment by tjaychen** [comment]
> It's possible I'm just being stupid, but I couldn't work out how to run the MUBI generation script. I also don't think they're wired into CI. This PR has two patches. The first adds some paths to the generation script so that it would run :-) The second runs it in CI.
> 
> @msfschaffner: Have I missed something obvious? If so, I'm very happy to drop this PR and do it the Right Way :-D

if you look under `util/design/`, there is a `gen-mubi.py`, that's the one i ran at least.

**Comment by rswarbrick** [comment]
> if you look under util/design/, there is a gen-mubi.py, that's the one i ran at least.

Oh! Ok, I'll take a look and delete one script or the other :-)

**Comment by tjaychen** [comment]
yeah that sounds good.

**Comment by rswarbrick** [comment]
@tjaychen: Would you mind taking another look? I don't think there's anything at `util/design/gen-mubi.py` and (assuming I've got the right Git incantation) I don't think there ever has been. What am I missing? :-/

File: hw/ip/prim/util/generate_prim_mubi.py (hunk 1 above)
Comment (rswarbrick) [reply hw/ip/prim/util/generate_prim_mubi.py:29]: Yeah, I could switch to using pathlib. Honestly, I think it sometimes makes things more messy (and you don't end up with such nice relative paths :-D). I'll push up a version doing that.

About the globals: could do, but it means more _SHOUTING_ABOUT_THINGS which I'm not a big fan of. It's not like we're running `main` lots of times, after all :-)

Comment (tjaychen) [reply hw/ip/prim/util/generate_prim_mubi.py:29]: btw, for determining repo top, should we use something like `git rev-parse --show-toplevel`? That way if we ever move the script around it seems like things will just stay stable.

**Comment by tjaychen** [comment]
> @tjaychen: Would you mind taking another look? I don't think there's anything at `util/design/gen-mubi.py` and (assuming I've got the right Git incantation) I don't think there ever has been. What am I missing? :-/

o crap...maybe i forgot to merge it... give me a sec...

**Comment by tjaychen** [comment]
> > @tjaychen: Would you mind taking another look? I don't think there's anything at `util/design/gen-mubi.py` and (assuming I've got the right Git incantation) I don't think there ever has been. What am I missing? :-/
> 
> o crap...maybe i forgot to merge it... give me a sec...

see #9176 . It does literally nothing other than calling the mubi generate stuff and format the C files.

**Review by tjaychen — APPROVED** [review]

**Pull request merged**